### DISCUSSION TITLE
two bim file display in one  view

### DISCUSSION THREAD

**Discussion by paprophet** [iTwin/itwinjs-core#1536] [Q&A] [answered]
when i open two bim  file display in one view ,
code like :https://github.com/imodeljs/imodeljs/blob/master/test-apps/display-test-app/src/frontend/TiledGraphics.ts
but i can not get select element 

![444](https://user-images.githubusercontent.com/18022720/120177512-1a989b80-c23b-11eb-869d-b2b26c758eb5.jpg)

**Comment by pmconne** [comment]
A viewport is associated with ONE iModel. Elements that tools interact with are identified by element Id, in the context of that ONE iModel.
You can write a TiledGraphicsProvider that draws elements from other iModels, but tools by default will not interact with those elements.
You must supply tools that opt-in to locating elements from other iModels, and can figure out what to do with them.
display-test-app's [selection tool](https://github.com/imodeljs/imodeljs/blob/0f65736b30fb8ebed071492659237c599faeb821/test-apps/display-test-app/src/frontend/App.ts#L74) allows locating elements from other iModels, but it assumes that the other iModel is just a different version of the viewport's iModel, so the element Ids in both refer to the same elements. This would not work if you display arbitrary other iModels in your viewport.

Perhaps you should specify why you are trying to display multiple iModels in a single viewport.

**Comment by paprophet** [comment]
by snapshot ,i must  change all the files(dgn,rvt and so on ) to a bim file ，if one small file of them have some change ,i must  change all the file to a bim file  again, and the snapshot process is very slow if the files is large.
so i want change the files to different bim files, and display them, if one file hava change, i just change this one file to bim file.

**Reply by paprophet** [reply]
so  i want the uiframework or base class can support imodelconnection object array .and
 now  just support one imodelconnection object.

**Reply by shenshengyi** [reply]
Hi, paprophet.
We found that you have asked a lot of questions about itwin.js in this forum, and you should be using itwin.js extensively. We can help you offline.
If you want, you can contact me, my email is Devin.Liu@bentley.com or WeChat luoshanji0926
Thank you.

**Comment by pmconne** [comment, marked as answer]
I am going to once again remind you that snapshots are for files that **are not changing**.
If your files are changing, you should be using iTwin Synchronizer, which **incrementally** updates an iModel based only on what changed between revisions of the source files.
The whole point of iModels is that they aggregate data from multiple sources and record change over time.
